frontend/src/APIClients/ResidentAPIClient.ts

```diff
@@ -67,8 +68,50 @@ const createResident = async ({
   }
 };
 
+const deleteResident = async (residentId: number): Promise<boolean> => {
```

Comment:
nit: We should probably follow the named parameters convention
```suggestion
const deleteResident = async ({residentId: number}): Promise<boolean> => {
```